safe_transaction_service/tokens/tasks.py

```diff
@@ -111,3 +113,33 @@ def fix_pool_tokens_task() -> Optional[int]:
         if number:
             logger.info("%d pool token names were fixed", number)
         return number
+
+
+@app.shared_task()
+@close_gevent_db_connection_decorator
+def update_token_info_from_token_list() -> int:
+    """
+    If there's at least one valid token list with at least 1 token, every token in the DB is marked as `not trusted`
+    and then every token on the is marked as `trusted`
+
+    :return: Number of tokens marked as `trusted`
+    """
+    tokens = []
+    for token_list in TokenList.objects.all():
+        tokens += token_list.get_tokens()
+
+    if not tokens:
+        return 0
+
+    # Make sure current chainId matches the one in the list
+    ethereum_network = get_ethereum_network()
+
+    token_addresses = [
+        fast_to_checksum_address(token["address"])
+        for token in tokens
+        if token["chainId"] == ethereum_network.value
+    ]
+
+    with transaction.atomic():
+        Token.objects.update(trusted=False)
+        return Token.objects.filter(address__in=token_addresses).update(trusted=True)
```

Comment:
Maybe instead of updating every `Token` we can update the ones where `address` is not in `token_addresses` to `False` and the other query set would update the other tokens to `True`. Maybe this can be done with query set exclusion.

This would result in less updates overall (items which are part of the list would get 1 update instead of 2).

Reply:
> Maybe instead of updating every `Token` we can update the ones where `address` is not in `token_addresses` to `False` and the other query set would update the other tokens to `True`. Maybe this can be done with query set exclusion.

Then you will force a query for every token not on the list, which is not quite fast (usually the "not in" SQL queries are not the best). I wouldn't do this, I think updating elements on the list 2 times is more optimal

Even though I updated `Token.objects.update(trusted=False)` to `Token.objects.filter(trusted=True).update(trusted=False)` to speed up the query

Reply:
> > Maybe instead of updating every `Token` we can update the ones where `address` is not in `token_addresses` to `False` and the other query set would update the other tokens to `True`. Maybe this can be done with query set exclusion.
> 
> Then you will force a query for every token not on the list, which is not quite fast (usually the "not in" SQL queries are not the best). I wouldn't do this, I think updating elements on the list 2 times is more optimal
> 
> Even though I updated `Token.objects.update(trusted=False)` to `Token.objects.filter(trusted=True).update(trusted=False)` to speed up the query

What about filtering for tokens on the list (updating those) and then excluding that set and update those to `False`. It'd still use `NOT()` at the end though... 🤔

Anyways was mostly an alternative that I thought 👍